x-pack/plugins/security_solution/public/common/components/sourcerer/index.tsx

```diff
@@ -43,27 +45,53 @@ interface SourcererComponentProps {
 export const Sourcerer = React.memo<SourcererComponentProps>(({ scope: scopeId }) => {
   const dispatch = useDispatch();
   const sourcererScopeSelector = useMemo(getSourcererScopeSelector, []);
-  const { configIndexPatterns, kibanaIndexPatterns, sourcererScope } = useSelector<
+  const { defaultDataView, kibanaDataViews, signalIndexName, sourcererScope } = useSelector<
     State,
     SourcererScopeSelector
   >((state) => sourcererScopeSelector(state, scopeId), deepEqual);
-  const { selectedPatterns, loading } = sourcererScope;
+  const { selectedDataViewId, selectedPatterns, loading } = sourcererScope;
   const [isPopoverOpen, setPopoverIsOpen] = useState(false);
+
+  const [dataViewId, setDataViewId] = useState<string>(selectedDataViewId ?? defaultDataView.id);
+
+  const { patternList, selectablePatterns } = useMemo(() => {
+    const theDataView = kibanaDataViews.find((dataView) => dataView.id === dataViewId);
+    return theDataView != null
+      ? {
+          patternList: theDataView.title
+            .split(',')
+            // remove duplicates patterns from selector
+            .filter((pattern, i, self) => self.indexOf(pattern) === i),
```

Comment:
do we really need to do that? might be useful to do it on the server side

Reply:
Yes. We need to show the index pattern title accurately in the dropdown so the user knows which index pattern it is, but then only offer one occurrence of each pattern in the combo box. You can see why that is necessary here:


<img width="631" alt="Screen Shot 2021-09-30 at 4 35 40 PM" src="https://user-images.githubusercontent.com/6935300/135539051-48203e13-c1b3-48e4-bc07-9daffc953745.png">
<img width="628" alt="Screen Shot 2021-09-30 at 4 35 46 PM" src="https://user-images.githubusercontent.com/6935300/135539005-4c38cbee-2575-4575-8619-960271f35595.png">

if i did the duplicate removal serverside:
<img width="626" alt="Screen Shot 2021-09-30 at 4 39 50 PM" src="https://user-images.githubusercontent.com/6935300/135539305-d3d6b80f-d069-487e-8646-0b119698be32.png">

Reply:
@XavierM i just added a `Quirks` section to the PR description, ill continue to update it with weirdness. if you see weird code not covered there, please let me know so i can add it